app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/ParkingView.java

```diff
@@ -15,8 +19,18 @@ public ParkingView(Activity activity, ActivityMainBinding binding) {
         this.binding = binding;
     }
 
+    @SuppressLint("StringFormatMatches")
     @Override
-    public void displayAvailableParkingSpots(String parkingSpots){
-        binding.parkingTextView.setText(getActivity().getString(R.string.activity_main_available_text) + parkingSpots);
+    public void displayAvailableParkingSpots(Long parkingSpots) {
+        binding.parkingTextView.setText(getActivity().getString(R.string.activity_main_available_text, parkingSpots));
+    }
+
+    @Override
+    public void displayFragment(ListenerDialogFragment listener) {
+        ParkingFragment fragment = ParkingFragment.newInstance(listener);
+        FragmentManager fragmentManager = getFragmentManager();
+        if (fragment != null) {
```

Comment:
this validation is not necessary as the newInstance method always return a not null object

Reply:
done